src/main/java/com/robobunny/SeekBarPreference.kt

```diff
@@ -0,0 +1,203 @@
+package com.robobunny
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.preference.Preference
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewParent
+import android.widget.LinearLayout
+import android.widget.SeekBar
+import android.widget.SeekBar.OnSeekBarChangeListener
+import android.widget.TextView
+import nl.mpcjanssen.simpletask.Logger
+import nl.mpcjanssen.simpletask.R
+
+
+class SeekBarPreference : Preference, OnSeekBarChangeListener {
+
+    private val TAG = javaClass.getName()
+    private val log: Logger
+
+    private var mMaxValue = 100
+    private var mMinValue = 0
+    private var mInterval = 1
+    private var mCurrentValue: Int = 0
+    private var mUnitsRight = ""
+    private var mSeekBar: SeekBar? = null
+
+    private var mStatusText: TextView? = null
+
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
+        log = Logger
+        initPreference(context, attrs)
+    }
+
+    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {
+        log = Logger
+        initPreference(context, attrs)
+    }
+
+    private fun initPreference(context: Context, attrs: AttributeSet) {
+        setValuesFromXml(attrs)
+        mSeekBar = SeekBar(context, attrs)
+        mSeekBar!!.max = mMaxValue - mMinValue
+        mSeekBar!!.setOnSeekBarChangeListener(this)
+
+        layoutResource = R.layout.seek_bar_preference

```

Comment:
First change: set the **layout** resource (the entire preference) instead of the **widget** resource (just the checkbox/switch/whatever).